### PR TITLE
ci: add stale bot

### LINKED ISSUE #2436: [Feature][operation] Add stale bot to our repo
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Some of our open issues had existed for over a year. Although many of them are valid, time had proven that their priorities are likely to be low.
These many issues in our repo make backlog management hard and time-consuming.

### Use case

1. Reduce the time of issue management
2. Help us focus on more important things
3. Remind us to revisit stale issues and re-evaluate them

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by klesh**
# Summary

Add stale bot

### Does this close any open issues?
Closes #2436

### CODE REVIEW

**Review by warren830 — APPROVED** [review]
LGTM

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+on:
+  schedule:
+    - cron: '0 0 * * *'
+permissions:
+  # Stale recommended permissions
+  pull-requests: write
+  issues: write
+jobs:
+  stale:
+    runs-on: ubuntu-latest
```
Comment (Startrekzky) [inline .github/workflows/stale.yml:28]: inline-comment-test1

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          # We do not stale Issues with label `Waiting for reply`, `new feature` and `DSIP`
+          exempt-issue-labels: 'Waiting for reply,new feature,DSIP'
+          stale-issue-message: >
+            This issue has been automatically marked as stale because it has not had recent activity
+            for 30 days. It will be closed in next 7 days if no further activity occurs.
+          close-issue-message: >
+            This issue has been closed because it has not received response for too long time. You could
+            reopen it if you encountered similar problems in the future.
+          # Stale PRs
+          days-before-pr-stale: 120
```
Comment (Startrekzky) [inline .github/workflows/stale.yml:44]: inline-comment-test2

**Review by Startrekzky — COMMENTED** [review]
A finished comment

**Comment by warren830** [comment]
test normal comment

**Comment by Startrekzky** [comment]
Normal comment - test

**Pull request merged**